### PR TITLE
[WIP] Logging

### PR DESCRIPTION
**Pull request by dotnetCarpenter**
This is my initial idea on how to provide logging for ecstatic.

The PR is only half done but the tests and main output is done.

TODO:
- [x] Tests
- [x] Documentation
- [ ] Refactor `middleware` to be able to log all status codes and response length
- [ ] Finish `getNscaDateTime` to handle all time zones

I think this is a candidate for a minor release.

Unfortunately, I don't have time to finish this, this year. If you @jfhbrook or someone else have time, I would love to review.

I noticed that the log code path is run two times for each request, which means that the middleware is run two times for each request but I don't understand why we would do that. The second time the `req.method` is `undefined` and `req.url` is encoded.

### CODE REVIEW

**Comment by dotnetCarpenter** [comment]
hehe failing in node 4 and 5 after I followed airBnB's eslint - dropping node 4 and 5 support to destructuring assignment or make two small fixes in tests... yaa, remove destructuring - it also sounds so negative.

File: tap-snapshots/test-cli.js-TAP.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* IMPORTANT
```
Comment (jfhbrook) [inline tap-snapshots/test-cli.js-TAP.test.js]: what is this :o

Comment (dotnetCarpenter) [reply tap-snapshots/test-cli.js-TAP.test.js]: http://www.node-tap.org/snapshots/

**Comment by dotnetCarpenter** [comment]
Ref: #208

**Comment by dotnetCarpenter** [comment]
@jfhbrook What do you think about the idea and about refactoring `middleware` - would you be up for doing it? Also do you know why the middleware is run twice for each request?

File: tap-snapshots/test-cli.js-TAP.test.js (hunk 1 above)
Comment (dotnetCarpenter) [reply tap-snapshots/test-cli.js-TAP.test.js]: I added a comment in the test about how to generate the file

**Comment by dotnetCarpenter** [comment]
Looks like codecov is broken somehow.. I'll take a quick look

**Comment by dotnetCarpenter** [comment]
I'm using this branch of ecstatic serve svg and image files to another node program and I never see double logging (re-run of `middleware` for each request), which leads me to believe that this only happens when requesting via a browser. I was using Chromium..
I'm guessing it could be OPTION or prefetch request that we are not handling very well. It's not completely b0rken but pretty bad for logging and probably has other side-effects.

**Comment by jfhbrook** [comment]
I know there's code branches that call the original middleware recursively, those would probably be candidates for ways to cause double-logging.

I'm down to do some middleware refactors in general, given I have the time and the understanding necessary to make useful ones. This might also touch on https://github.com/jfhbrook/node-ecstatic/issues/193 https://github.com/jfhbrook/node-ecstatic/issues/154 https://github.com/jfhbrook/node-ecstatic/issues/88

**Comment by dotnetCarpenter** [comment]
R. I. P.
#259

**Pull request closed without merging**